### PR TITLE
updates to application objectives & summary

### LINKED ISSUE #116: Application of a modular workflow
Lesson Summary
- Use the packages and some of their functions to create a common data analysis workflow in a modular and reproducible manner. Then, see how this approach handles common scenarios when compared to traditional analyses.
- What is modularity and why is it more reproducible?

Lesson Objectives
By the end of this lesson, the learner will be able to:
1. Automate retrieval and plotting of data.
2. Use USGS packages within an overall data analysis workflow (apply modularity).
3. Contribute to and seek help within the USGS-R community.

Lesson Assessment
None

`9999-06-01`

### PR DESCRIPTION
**Pull request by lindsaycarr**
Fixes #116

### CODE REVIEW

File: content/usgs-packages/Application_Intro.Rmd [hunk 1]
```diff
@@ -32,14 +32,16 @@ set.seed(1)
 
 ## Lesson Summary
 
-TBD
+This lesson will revisit the packages taught in previous lessons and use some of their functions to create a common data analysis workflow in a modular and reproducible manner. Then, you can learn how a modular approach compares to traditional analyses.
+
+What is modularity and why is it more reproducible? Modularity is the idea that your workflow leverages other tools to do some of the more general tasks. You can think of it as a sort of "plug-and-play" scenario, where you are relying on functionality from another developer to perform pieces of your analysis. This type of workflow reduces some of the maintenance burden because you are pushing some of the responsibility for code accuracy and reliability onto the tools. This also means that you should choose your tools wisely and scrutinize the tools you plan to use in your analyses.
```
Comment (aappling-usgs) [inline content/usgs-packages/Application_Intro.Rmd]: how about '"plug-and-play" approach' instead of '"plug-and-play" scenario'?

Comment (aappling-usgs) [inline content/usgs-packages/Application_Intro.Rmd]: i don't think you've answered your question of why modularity is more reproducible.

File: content/usgs-packages/Application_Intro.Rmd [hunk 2]
```diff
@@ -32,14 +32,16 @@ set.seed(1)
...
 ## Lesson Objectives
 
 Use the packages and some of their functions to create a common data analysis workflow in a modular and reproducible manner.
 
 By the end of this lesson, the learner will be able to:
 
-1. Recognize similarities and differences in gsplot and base R plotting.
-2. Construct piped plotting calls.
-3. Create customized style files.
+1. Automate retrieval of data via NWIS, ScienceBase, and the GDP.
```
Comment (aappling-usgs) [inline content/usgs-packages/Application_Intro.Rmd]: because it's "via", maybe specify NWIS Web (the platform) in contrast to NWIS (the underlying data source). Might matter to some of our target audience

File: content/usgs-packages/Application_Intro.Rmd [hunk 3]
```diff
@@ -32,14 +32,16 @@ set.seed(1)
...
+2. Use USGS packages within an overall data analysis workflow (apply modularity).
```
Comment (aappling-usgs) [inline content/usgs-packages/Application_Intro.Rmd]: reproduciblity isn't called out in this list. maybe it should be? maybe in point 1, associated with 'automate'?

File: content/usgs-packages/Application_Intro.Rmd [hunk 4]
```diff
@@ -32,14 +32,16 @@ set.seed(1)
 
 ## Lesson Summary
 
-TBD
+This lesson will revisit the packages taught in previous lessons and use some of their functions to create a common data analysis workflow in a modular and reproducible manner. Then, you can learn how a modular approach compares to traditional analyses.
```
Comment (aappling-usgs) [inline content/usgs-packages/Application_Intro.Rmd]: this is an intriguing framing for the lesson. i'm looking forward to seeing how the rest of it plays out. is it worth noting here that everybody uses some modularity all the time? we use Excel, Word, Google sites...all these things are somebody else's software tool that helps us do our own work. i guess the difference is that somebody might be tempted to create their own URLs for an API call to NWIS Web, for example, whereas they'd never think/bother to try to build their own version of Word. anyway, i'm thinking it might be hard to follow through on the promise in the second sentence above...see how it pans out, then revise this if needed.

File: content/usgs-packages/Application_Intro.Rmd (hunk 1 above)
Comment (lindsaycarr) [inline content/usgs-packages/Application_Intro.Rmd]: Hmmm I think I was trying to allude to  that with talking about reliability and maintenance burden. Now I'm struggling to think of the relationship - does being modular make you more reproducible, or are they two complimentary practices?

File: content/usgs-packages/Application_Intro.Rmd (hunk 2 above)
Comment (lindsaycarr) [inline content/usgs-packages/Application_Intro.Rmd]: Ah yes, I should be explicit.

File: content/usgs-packages/Application_Intro.Rmd (hunk 4 above)
Comment (lindsaycarr) [inline content/usgs-packages/Application_Intro.Rmd]: I like this idea - I think it would help people who are not familiar understand the concept of modularity a bit better.

Not sure what you mean by the "promise" from the second sentence?

File: content/usgs-packages/Application_Intro.Rmd (hunk 1 above)
Comment (aappling-usgs) [inline content/usgs-packages/Application_Intro.Rmd]: i think it's a little of both. the biggest part of what makes it reproducible is the fact that it's scripted at all. and then also, in an ideal world tool maintainers keep their tools up to date with the databases and software dependencies they use, so that your own code will keep running for more months/years after you wrote it, so in that sense modularity -> reproducibility. and then there's a little bit of reproducibility coming from the fact that someone else could look at your code and understand it better if you're outsourcing some complexity to other tools, so they could reproduce or even modify your script more readily. but still, i'd say 80% of reproducibility is complementary to, rather than stemming from, modularity.

File: content/usgs-packages/Application_Intro.Rmd (hunk 4 above)
Comment (aappling-usgs) [inline content/usgs-packages/Application_Intro.Rmd]: i mean that you're promising to compare modular to traditional. are you going to show a script that implements a geoknife extraction/downloading process using base R only? sounds scary to me. i really just made this comment to make sure you were thinking about it - i'm content to wait until i see the rest of the lesson before you make any particular edits here.

Comment (lindsaycarr) [inline content/usgs-packages/Application_Intro.Rmd]: I guess I wasn't really thinking hard about my wording there. I was thinking of more of an internal comparison - where the user goes "huh, that is simpler"

Comment (lindsaycarr) [inline content/usgs-packages/Application_Intro.Rmd]: I should probably reword if that sentence made you think of showing the base R implementation.

File: content/usgs-packages/Application_Intro.Rmd (hunk 1 above)
Comment (lindsaycarr) [inline content/usgs-packages/Application_Intro.Rmd]: OK mulling this over and thinking of what language to use to capture this.....I'll push up some changes later

**Comment by lindsaycarr** [comment]
@aappling-usgs could you just give a final look at the new language I used based on your feedback?

**Comment by aappling-usgs** [comment]
oh, shoot, sorry! didn't realize i was leaving you hanging on this PR. yes, will look right away

**Comment by lindsaycarr** [comment]
@aappling-usgs I was letting your feedback hang - just pushed up the changes ~2min ago :)

**Comment by aappling-usgs** [comment]
and oh, phew =)
